Hakizimana-Clement/Advizr
# [Feature]: Implement Advice API Fetching with React Query and Button Animation

#### PR Description
This PR integrates an external API to fetch random advice and manages data retrieval using React Query for optimized caching and state management. 

It also adds an animation effect to the button, enhancing the user experience by visually indicating the fetching of new advice.

### What does this PR do?
This PR introduces a new feature that fetches advice data from an external API, leverages React Query for data management, and adds an animation to the "Get Advice" button.

### Description of Task to be Completed
- [x] Task 1: Install and configure React Query for state and caching management.
- [x] Task 2: Create a function to fetch random advice data from the API.
- [x] Task 3: Use React Query to call the API function and manage the fetched advice data.
- [x] Task 4:  Implement button animation to indicate loading each time a new piece of advice is requested.

### How Should This Be Manually Tested?
1. Clone the repository and navigate into the project directory.
2. Check out the branch **ft-api** and install dependencies with **npm install.**
3. Start the application locally by running npm run dev.
4. On the homepage, click the "Get Advice" button to load new advice. Observe the animation while data is being fetched.

### Any Background Context You Want to Provide?
The react-query and react-spinners packages were added to streamline data fetching, caching, and loading animations, enhancing the user experience during API calls.

### What Are the Relevant Pivotal Tracker Stories?
Feature #6
plement Advice API Fetching with React Query and Button Animation

### Related Issues
Issue #7
